proto/iotics/api/host.proto

```diff
@@ -23,14 +23,22 @@ option php_namespace = "Iotics\\Api";
 service HostAPI {
   // GetHostID gets the ID of the host twin.
   rpc GetHostID(GetHostIDRequest) returns (GetHostIDResponse) {}
+
+  // IsHostDataAllowed determines whether a remote host is allowed to perform data requests against the given twin.
+  // An example of a data request is interest.SendInputMessage
+  rpc IsHostDataAllowed(IsHostDataAllowedRequest) returns (IsHostDataAllowedResponse) {}
```

Comment:
Note: The slight change to the RPC naming compared to what's in the ticket is intentional: I thought since the ontological terms talk about e.g. `pub:hostMetadataAllowList`, it might be more consistent to call it `IsHost<Type>Allowed()`